core/src/main/java/org/elasticsearch/cluster/routing/allocation/command/AllocationCommand.java

```diff
@@ -21,59 +21,24 @@
 
 import org.elasticsearch.cluster.routing.allocation.RerouteExplanation;
 import org.elasticsearch.cluster.routing.allocation.RoutingAllocation;
-import org.elasticsearch.common.Nullable;
-import org.elasticsearch.common.io.stream.StreamInput;
-import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.common.io.stream.NamedWriteable;
 import org.elasticsearch.common.xcontent.ToXContent;
-import org.elasticsearch.common.xcontent.XContentBuilder;
 import org.elasticsearch.common.xcontent.XContentParser;
 
 import java.io.IOException;
 
 /**
  * This interface defines the basic methods of commands for allocation
  */
-public interface AllocationCommand {
-
-    /**
-     * Factory to create {@link AllocationCommand}s
-     * @param <T> Type of {@link AllocationCommand}s created by this {@link Factory}
-     */
-    interface Factory<T extends AllocationCommand> {
-
-        /**
-         * Reads an {@link AllocationCommand} of type <code>T</code> from a {@link StreamInput}
-         * @param in {@link StreamInput} to read the {@link AllocationCommand} from
-         * @return {@link AllocationCommand} read from the {@link StreamInput}
-         * @throws IOException if something happens during reading
-         */
-        T readFrom(StreamInput in) throws IOException;
-
+public interface AllocationCommand extends NamedWriteable<AllocationCommand>, ToXContent {
+    interface Parser<T extends AllocationCommand> {
```

Comment:
To be clear - I'd like to do something like this but maybe not as part of this PR.